### PR TITLE
LIMS-4267: edit order material type 

### PR DESCRIPTION
**Pull request by ghada-37**
Apply this on suborder number 5 for example:-
        -When user update the materiel type from table view once I delete it message will appear
        (All analysis created with this order and test plan/ test unit will be deleted )
        -Once you press on OK button, the material type & article & test pan/ test unit will delete
        -You can update it by choose another one and choose corresponding article & test plan/ test unit

        LIMS-4264 ( order with test plan)
        LIMS-4267 (order with test unit )

```
➜  1lims-automation git:(LIMS-4267) nosetests -vs --nologcapture --tc-file=config.ini ui_testing/testcases/basic_tests/test06_orders.py:OrdersTestCases.test027_update_material_type_0_testPlans
2020-06-09 02:45:17.109 | INFO     | api_testing.apis.base_api:info:76 - Get authorized api session.
2020-06-09 02:45:19.697 | INFO     | api_testing.apis.base_api:info:76 - session ID : eyJhbGciOi .....
Apply this on suborder number 5 for example:- [with case='testPlans'] ...       
2020-06-09 02:45:19.888 | INFO     | ui_testing.testcases.base_test:info:135 - Test case : test027_update_material_type_0_testPlans
2020-06-09 02:45:49.396 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:537 - wait until page is loaded
2020-06-09 02:45:50.850 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orders/auto/generatedId
2020-06-09 02:45:52.990 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:45:52.991 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans
2020-06-09 02:45:54.300 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:45:54.301 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/materialTypes
2020-06-09 02:45:56.072 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:45:56.073 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans/get/833
2020-06-09 02:45:56.538 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:45:56.539 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/contacts
2020-06-09 02:45:57.381 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:45:57.382 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orders
2020-06-09 02:45:58.274 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:45:58.275 | INFO     | ui_testing.testcases.base_test:info:135 - new order created with order data [{'orderNo': 9075244500, 'contact': [{'id': 458, 'text': 'f82f3f2acc', 'No': '32750'}], 'deletedTestPlans': [], 'deletedAnalysisIds': [], 'dynamicFieldsValues': [], 'analysisNo': [], 'selectedDepartments': [], 'orderType': {'id': 1, 'text': 'New Order'}, 'departments': [], 'attachments': [], 'testPlans': [{'id': 1005, 'name': '3e3fb1ec38', 'version': 2}], 'selectedTestPlans': [{'id': 1005, 'name': '3e3fb1ec38', 'version': 2}], 'testUnits': [{'id': 1006, 'name': '3e3fb1ec38', 'new': True}], 'selectedTestUnits': [{'id': 1006, 'name': '3e3fb1ec38', 'new': True}], 'materialType': {'id': 1, 'text': 'Raw Material'}, 'materialTypeId': 1, 'article': {'id': 1462, 'text': '68ad1a371e'}, 'articleId': 1462, 'shipmentDate': '2020-06-09', 'testDate': '2020-06-09', 'year': '2020', 'yearOption': 1, 'shipmentDatedateOption': {'year': '2020', 'month': '06', 'day': '09'}, 'testDatedateOption': {'year': '2020', 'month': '06', 'day': '09'}, 'orderNoWithYear': '9075244500-2020'}]
2020-06-09 02:45:58.275 | INFO     | ui_testing.testcases.base_test:info:135 - get random completed test plan with different material type
2020-06-09 02:45:58.276 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans
2020-06-09 02:46:00.315 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:46:00.316 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans/get/790
2020-06-09 02:46:00.689 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:46:00.690 | INFO     | ui_testing.testcases.base_test:info:135 - update material type of order from Raw Material to TestArchive
2020-06-09 02:46:02.241 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-06-09 02:46:15.163 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-06-09 02:46:20.227 | INFO     | ui_testing.testcases.base_test:info:135 - confirm pop_up
2020-06-09 02:46:20.227 | INFO     | ui_testing.pages.base_pages:confirm_popup:79 - confirming the popup
2020-06-09 02:46:20.444 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-06-09 02:46:25.449 | INFO     | ui_testing.testcases.base_test:info:135 - assert article and test plan/ test unit  are empty
2020-06-09 02:46:35.450 | INFO     | ui_testing.testcases.base_test:info:135 - set article to a6eaed14e9 and test plan to 44d2616728
2020-06-09 02:46:43.026 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-06-09 02:46:48.031 | INFO     | ui_testing.pages.base_pages:info:280 -  save the changes
2020-06-09 02:46:48.075 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-09 02:46:50.308 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-09 02:46:52.310 | INFO     | ui_testing.pages.base_pages:info:280 -  refresh to make sure that data are saved correctly
2020-06-09 02:46:59.093 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:537 - wait until page is loaded
2020-06-09 02:46:59.139 | INFO     | ui_testing.testcases.base_test:info:135 - get order data after edit and refresh
2020-06-09 02:46:59.139 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orderInformation/get/orders/byId/2087
2020-06-09 02:46:59.853 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:46:59.854 | INFO     | ui_testing.testcases.base_test:info:135 - navigate to analysis page to make sure analysis corresponding to suborder updated
2020-06-09 02:47:01.130 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:537 - wait until page is loaded
2020-06-09 02:47:04.523 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:537 - wait until page is loaded
2020-06-09 02:47:10.309 | INFO     | ui_testing.pages.base_pages:info:280 -  filter by analysis number : 2937-20
2020-06-09 02:47:10.309 | INFO     | ui_testing.pages.base_pages:open_filter_menu:88 -  Open Filter
2020-06-09 02:47:19.380 | INFO     | ui_testing.testcases.base_test:info:135 - TearDown.        
ok

----------------------------------------------------------------------
Ran 1 test in 119.494s

OK


➜  1lims-automation git:(LIMS-4267) ✗  nosetests -vs --nologcapture --tc-file=config.ini ui_testing/testcases/basic_tests/test06_orders.py:OrdersTestCases.test027_update_material_type_1_testUnit
2020-06-09 02:48:23.753 | INFO     | api_testing.apis.base_api:info:76 - Get authorized api session.
2020-06-09 02:48:24.348 | INFO     | api_testing.apis.base_api:info:76 - session ID : eyJhbGciOi .....
Apply this on suborder number 5 for example:- [with case='testUnit'] ...        
2020-06-09 02:48:24.553 | INFO     | ui_testing.testcases.base_test:info:135 - Test case : test027_update_material_type_1_testUnit
2020-06-09 02:48:57.231 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:537 - wait until page is loaded
2020-06-09 02:48:58.518 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orders/auto/generatedId
2020-06-09 02:48:59.111 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:48:59.111 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans
2020-06-09 02:49:00.402 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:49:00.404 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/materialTypes
2020-06-09 02:49:00.733 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:49:00.734 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans/get/1153
2020-06-09 02:49:01.112 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:49:01.113 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/contacts
2020-06-09 02:49:02.114 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:49:02.115 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orders
2020-06-09 02:49:02.770 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:49:02.770 | INFO     | ui_testing.testcases.base_test:info:135 - new order created with order data [{'orderNo': 9075244502, 'contact': [{'id': 206, 'text': '47588d3057', 'No': '1be0604b83'}], 'deletedTestPlans': [], 'deletedAnalysisIds': [], 'dynamicFieldsValues': [], 'analysisNo': [], 'selectedDepartments': [], 'orderType': {'id': 1, 'text': 'New Order'}, 'departments': [], 'attachments': [], 'testPlans': [{'id': 1319, 'name': 'b37baa0bf3', 'version': 1}], 'selectedTestPlans': [{'id': 1319, 'name': 'b37baa0bf3', 'version': 1}], 'testUnits': [{'id': 12, 'name': 'tu_all_mb', 'new': True}], 'selectedTestUnits': [{'id': 12, 'name': 'tu_all_mb', 'new': True}], 'materialType': {'id': 109, 'text': 'ed3fdc710d'}, 'materialTypeId': 109, 'article': {'id': 1597, 'text': '9979c132f7'}, 'articleId': 1597, 'shipmentDate': '2020-06-09', 'testDate': '2020-06-09', 'year': '2020', 'yearOption': 1, 'shipmentDatedateOption': {'year': '2020', 'month': '06', 'day': '09'}, 'testDatedateOption': {'year': '2020', 'month': '06', 'day': '09'}, 'orderNoWithYear': '9075244502-2020'}]
2020-06-09 02:49:02.771 | INFO     | ui_testing.testcases.base_test:info:135 - get random completed test plan with different material type
2020-06-09 02:49:02.771 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans
2020-06-09 02:49:04.011 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:49:04.013 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans/get/441
2020-06-09 02:49:04.353 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:49:04.354 | INFO     | ui_testing.testcases.base_test:info:135 - update material type of order from ed3fdc710d to abd299009a
2020-06-09 02:49:05.849 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-06-09 02:49:18.842 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-06-09 02:49:23.888 | INFO     | ui_testing.testcases.base_test:info:135 - confirm pop_up
2020-06-09 02:49:23.888 | INFO     | ui_testing.pages.base_pages:confirm_popup:79 - confirming the popup
2020-06-09 02:49:24.091 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-06-09 02:49:29.093 | INFO     | ui_testing.testcases.base_test:info:135 - assert article and test plan/ test unit  are empty
2020-06-09 02:49:39.212 | INFO     | ui_testing.testcases.base_test:info:135 - set article to 90ce965031 and test unit to fa4dfa3f98
2020-06-09 02:49:46.752 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-06-09 02:49:51.755 | INFO     | ui_testing.pages.base_pages:info:280 -  save the changes
2020-06-09 02:49:51.815 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-09 02:49:54.057 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-09 02:49:56.058 | INFO     | ui_testing.pages.base_pages:info:280 -  refresh to make sure that data are saved correctly
2020-06-09 02:50:01.079 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:537 - wait until page is loaded
2020-06-09 02:50:01.122 | INFO     | ui_testing.testcases.base_test:info:135 - get order data after edit and refresh
2020-06-09 02:50:01.122 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orderInformation/get/orders/byId/2090
2020-06-09 02:50:01.499 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-09 02:50:01.500 | INFO     | ui_testing.testcases.base_test:info:135 - navigate to analysis page to make sure analysis corresponding to suborder updated
2020-06-09 02:50:02.850 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:537 - wait until page is loaded
2020-06-09 02:50:05.930 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:537 - wait until page is loaded
2020-06-09 02:50:09.953 | INFO     | ui_testing.pages.base_pages:info:280 -  filter by analysis number : 2939-20
2020-06-09 02:50:09.953 | INFO     | ui_testing.pages.base_pages:open_filter_menu:88 -  Open Filter
2020-06-09 02:50:19.161 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-06-09 02:50:30.000 | INFO     | ui_testing.testcases.base_test:info:135 - TearDown.        
ok

----------------------------------------------------------------------
Ran 1 test in 125.447s

OK


```

### CODE REVIEW

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 1]
```diff
@@ -1076,66 +1076,74 @@ def test024_archive_sub_order(self):
...
+        -Once you press on OK button, the material type & article & test pan/ test unit will delete
+        -You can update it by choose another one and choose corresponding article & test plan/ test unit
 
+        LIMS-4264 ( order with test plan )
+
+        LIMS-4267 (order with test unit )
+        """
+        self.info('create new order')
+        api, order_payload = self.orders_api.create_new_order()
+        self.assertEqual(api['status'], 1, api)
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test06_orders.py:1093]:
```suggestion
        self.assertEqual(api['status'], 1, order_payload)
```

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 2]
```diff
@@ -1076,66 +1076,74 @@ def test024_archive_sub_order(self):
...
+        self.info('new order created with order data {}'.format(order_payload))
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test06_orders.py:1094]: remove it

Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py:1094]: line 1094 only right ? as you selected multiple lines

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 1 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py]: done

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 2 above)
Comment (0xIslamTaha) [reply ui_testing/testcases/basic_tests/test06_orders.py:1094]: yes only 1094

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 3]
```diff
@@ -1076,66 +1076,74 @@ def test024_archive_sub_order(self):
...
+        (All analysis created with this order and test plan/ test unit will be deleted )
+        -Once you press on OK button, the material type & article & test pan/ test unit will delete
+        -You can update it by choose another one and choose corresponding article & test plan/ test unit
 
+        LIMS-4264 ( order with test plan )
+
+        LIMS-4267 (order with test unit )
+        """
+        self.info('create new order')
+        api, order_payload = self.orders_api.create_new_order()
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test06_orders.py:1092]:
```suggestion
        response, order_payload = self.orders_api.create_new_order()
```

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 2 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py]: done

File: api_testing/apis/test_plan_api.py [hunk 4]
```diff
@@ -271,3 +271,14 @@ def get_testunits_in_testplan_by_No(self, no):
         test_units = self.get_testunits_in_testplan(test_plan_id)
         test_units_names = [testunit['name'] for testunit in test_units]
         return test_units_names
+
+    def get_suborder_data_with_different_material_type(self, material_type):
+        test_plans = self.get_completed_testplans(limit=1000)
+        # I need to make sure that material type not equal '47d56b4399' due to this open bug
+        # https://modeso.atlassian.net/browse/LIMS-7710
+        test_plans_without_duplicate = [test_plan for test_plan in test_plans if test_plan['materialType']
+                                        not in [material_type, '47d56b4399']]
```
[the remark this answers is not in the document]
Comment (ghada-37) [reply api_testing/apis/test_plan_api.py:282]: ok

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 5]
```diff
@@ -1076,66 +1076,74 @@ def test024_archive_sub_order(self):
...
-        test_plan = random.choice(test_plans_without_duplicate)
-
-        self.info('update material type of order {} with {}'.format(order['orderNo'], test_plan['materialType']))
-        self.orders_page.get_order_edit_page_by_id(order['id'])
-        self.order_page.update_suborder(sub_order_index=suborder_update_index, material_type=test_plan['materialType'])
-        confirm_edit = self.base_selenium.check_element_is_exist(element="general:confirmation_pop_up")
-        confirm_edit_message = self.base_selenium.get_text(element="general:confirmation_pop_up")
-        self.assertTrue(confirm_edit)
-        self.assertIn('All analysis created with this order and test plan will be deleted', confirm_edit_message)
+        test_plan,  test_unit = TestPlanAPI().get_suborder_data_with_different_material_type(
```
[the remark this answers is not in the document]
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py]: done

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 3 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py]: done

File: api_testing/apis/test_plan_api.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ghada-37) [reply api_testing/apis/test_plan_api.py:282]: image
I couldn't delete it

**Pull request merged**